### PR TITLE
Merge ChanJianHao-V2-DG-Workspace-Updates

### PR DESCRIPTION
**Pull request by ChanJianHao**
Updated DG with Workspace writeup and diagrams.
Refactored some of my code.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -436,12 +436,122 @@ Given below is an example of the usage scenario of view information command and
...
+
+> :memo: The other options (`-s`, `-l`, `-d`) follows a similar process, only the list and switch option does not interact with `StorageManager` and `Watchlist`.
+
+![Workspace Command Sequence Diagram](images/WorkspaceCommand-Sequence-Diagram.png) <br/>
+*Figure 18: Workspace Command After New Workspace Delete*
+
+
+
+<br/>
+
```
Comment (OngDeZhi) [inline docs/DeveloperGuide.md]: Is there a reason for the chunk of space between the end of 4.4.1 and the header of 4.4.2?

**Review by OngDeZhi — COMMENTED** [review]
Besides the one I mentioned, it looks good!

**Review by EyoWeiChin — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ChanJianHao) [reply docs/DeveloperGuide.md]: Good catch, removed them!

**Review by OngDeZhi — APPROVED** [review]
LGTM.

**Pull request merged**